app/src/main/java/com/example/nvdovin/weatherapp/presentation/details/adapter/MainRecyclerAdapter.java

```diff
@@ -0,0 +1,134 @@
+
+    private static final int FORECAST_CARD = 0;
+    private static final int PARAMETERS_CARD = 1;
+    private static final int NUMBER_OF_CARDS_TO_DISPLAY = 2;
+    private List<DailyForecast> dailyForecastList;
+    private ImageUtils imageUtils;
+    private WeatherData currentWeatherData;
+    private Context context;
+
+
+    public MainRecyclerAdapter(Context context, List<DailyForecast> dailyForecastList, WeatherData currentWeatherData) {
+        this.dailyForecastList = dailyForecastList;
+        imageUtils = new ImageUtils(context);
+        this.currentWeatherData = currentWeatherData;
+        this.context = context;
+    }
+
+    @Override
+    public ViewHolder<DailyForecast> onCreateViewHolder(ViewGroup parent, int viewType) {
+        ViewHolder viewHolder;
+        switch (viewType) {
+            case FORECAST_CARD:
+                viewHolder = new ForecastCard(LayoutInflater
+                        .from(parent.getContext()).inflate(R.layout.detail_forecast_card, parent, false));
+                break;
+            case PARAMETERS_CARD:
+                viewHolder = new ParametersCard(LayoutInflater
+                        .from(parent.getContext()).inflate(R.layout.detail_parameters_card, parent, false));
+                break;
+            default:
+                viewHolder = new ForecastCard(LayoutInflater
+                        .from(parent.getContext()).inflate(R.layout.detail_forecast_card, parent, false));
+        }
+        return viewHolder;
+    }
+
+
+    @Override
+    public void onBindViewHolder(ViewHolder<DailyForecast> holder, int position) {
+        holder.bindData(dailyForecastList.get(position));
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return position;
+    }
+
+    @Override
+    public int getItemCount() {
+        return NUMBER_OF_CARDS_TO_DISPLAY;
+    }
+
+    class ForecastCard extends ViewHolder<DailyForecast> {
+        @BindView(R.id.detail_forecast_today_recycler)
+        RecyclerView todayDataRecycler;
+        @BindView(R.id.detail_forecast_next_days_recycler)
+        RecyclerView nextDaysRecycler;
+
+        ForecastCard(View itemView) {
+            super(itemView);
+        }
+
+        @Override
+        public void bindData(DailyForecast dailyForecast) {
+            Context context = itemView.getContext();
+            todayDataRecycler.setLayoutManager(new LinearLayoutManager
+                    (context, LinearLayoutManager.HORIZONTAL, false));
+            todayDataRecycler.setAdapter(new TodayForecastAdapter(dailyForecast, context));
+
+            nextDaysRecycler.setLayoutManager(new LinearLayoutManager(context));
+            nextDaysRecycler.setNestedScrollingEnabled(false);
+            nextDaysRecycler.setAdapter(new NextDaysRecyclerAdapter(dailyForecastList, context));
+        }
+    }
+
+    class ParametersCard extends ViewHolder<DailyForecast> {
```

Comment:
I guess a better name for that will be ParametersCardHolder if it represents a specific kind of ViewHolder.